src/common/buffer.cc

```diff
@@ -1324,8 +1325,14 @@ static ceph::spinlock debug_lock;
     // make a new buffer.  fill out a complete page, factoring in the
     // raw_combined overhead.
     size_t need = round_up_to(len, sizeof(size_t)) + sizeof(raw_combined);
-    size_t alen = round_up_to(need, CEPH_BUFFER_ALLOC_UNIT) -
-      sizeof(raw_combined);
+    size_t alen = round_up_to(need, CEPH_BUFFER_ALLOC_UNIT_BASE);
+    if (_carriage == &_buffers.back()) {
+      size_t nlen = (_carriage->raw_length() + sizeof(raw_combined)) << 1;
```

Comment:
`... << 1`

Maybe just `... * 2` and let compiler to optimize?

Reply:
my brain likes bitshift in this scenario as I find it gives me more context regarding the goal, but I'm open if you don't like it. :)